src/routes/formsg/formsgGGsRepair.ts

```diff
@@ -0,0 +1,264 @@
+      if (!attachmentId) {
+        throw new Error("No attachment id")
+      }
+      const decryptedFile: DecryptedFile =
+        res.locals.submission.attachments?.[attachmentId]
+      const reposCsv = Buffer.from(decryptedFile.content).toString()
+      if (!reposCsv.startsWith("repo_name")) {
+        logger.error("Invalid csv format")
+        return
+      }
+      const repos = reposCsv.split("\n").slice(1)
+      repos.forEach((repo) => {
+        repoNames.push(repo)
+      })
+      if (repoNames.length === 0) {
+        logger.error("No repo name provided")
+        return
+      }
+    } else {
+      const repoNamesFromTable = getFieldsFromTable(responses, REPO_NAME_FIELD)
+
+      if (!repoNamesFromTable) {
+        logger.error("No repo name provided")
+        return
+      }
+      repoNamesFromTable.forEach((repoName) => {
+        if (typeof repoName === "string") {
+          // actually wont happen based on our formsg form, but this code
+          // is added defensively
+          repoNames.push(repoName)
+        } else {
+          repoNames.push(repoName[0])
+        }
+      })
+    }
+
+    if (!requesterEmail?.endsWith("@open.gov.sg")) {
+      logger.error("Requester email is not from @open.gov.sg")
+      return
+    }
+    this.handleGGsFormSubmission(repoNames, requesterEmail)
+  }
+
+  handleGGsFormSubmission = (repoNames: string[], requesterEmail: string) => {
+    const repairs: ResultAsync<string, GitFileSystemError>[] = []
+
+    const clonedStagingRepos: string[] = []
+    const syncedStagingAndStagingLiteRepos: string[] = []
+    repoNames.forEach((repoName) => {
+      const repoUrl = `git@github.com:isomerpages/${repoName}`
+
+      repairs.push(
+        this.doesRepoNeedClone(repoName)
+          .andThen(() => {
+            const isStaging = true
+            return (
+              this.gitFileSystemService
+                // Repo does not exist in EFS, clone it
+                .cloneBranch(repoName, isStaging)
+                .andThen(() => {
+                  // take note of repos that cloned successfully
+                  clonedStagingRepos.push(repoName)
+                  return okAsync(true)
+                })
+            )
+          })
+          .orElse((error) => {
+            if (error === false) {
+              // Repo exists in EFS, no need to clone, but continue with syncing
+              return okAsync(false)
+            }
+            return errAsync<boolean, GitFileSystemError>(error)
+          })
+          .andThen(() =>
+            // repo exists in efs, but we need to pull for staging and reset staging lite
+            this.gitFileSystemService
+              .pull(repoName, "staging")
+              .andThen(() =>
+                fromPromise(
+                  this.reposService.setUpStagingLite(
+                    path.join(EFS_VOL_PATH_STAGING_LITE, repoName),
+                    repoUrl
+                  ),
```

Comment:
Hmm actually does this assume that a staging lite doesn't exist? it seems to attempt a clone + pull, but we also want this to be able to repair error right, is this the desired behaviour?

Reply:
actually fair this not v clear, updated pr desc 
We always assume staging lite doesnt exist. the reason why we dont git pull is because once we git pull, .git reconciles and the folder bloats, and we lose the benefits of a small bundle size. so we always reset, and this is not too long (< 1min for nlb iirc.

We also always clone and pull. the reason for this is because cloning takes time, with repos like nlb it can take a while, so we do a pull right before we reset staging lite.

It is notable that during this period of time, any changes to the repo will screw this repair up, but actually this is ok because:

we can always re-run this script
we are not rate-limited in any way for the above ^